lib/overcommit/git_repo.rb

```diff
@@ -248,5 +248,16 @@ def submodules(options = {})
       raise Overcommit::Exceptions::GitSubmoduleError,
             "Unable to read submodule information from #{ref}:.gitmodules file: #{ex.message}"
     end
+
+    # Returns the names of all branches containing the given commit.
+    #
+    # @param commit_ref [String] git tree ref that resolves to a commit
+    # @return [Array<String>] list of branches containing the given commit
+    def branches_containing_commit(commit_ref)
+      `git branch --contains #{commit_ref}`.
+        split("\n").
+        reject { |s| s[/detached from/] }.  # ignore detached HEAD
+        map { |s| s[2..-1] }                # trim leading 2 characters
```

Comment:
Ah, wasn't aware of that option. Still need to snip out the `(detached from <hash>)` entry before splitting on whitespace, though.